src/lib/mongoBackend/mongoListSubscriptions.cpp

```diff
@@ -23,42 +23,224 @@
+*
+* setSubscriptionId -
+*/
+extern void setSubscriptionId(Subscription* s, const BSONObj& r)
+{
+  s->id = r.getField("_id").OID().toString();
+}
+
+
+/* ****************************************************************************
+*
+* setSubject -
+*/
+extern void setSubject(Subscription* s, const BSONObj& r)
+{
+  // Entities
+  std::vector<BSONElement> ents = r.getField(CSUB_ENTITIES).Array();
+  for (unsigned int ix = 0; ix < ents.size(); ++ix)
+  {
+    BSONObj ent           = ents[ix].embeddedObject();
+    std::string id        = ent.getStringField(CSUB_ENTITY_ID);
+    std::string type      = ent.getStringField(CSUB_ENTITY_TYPE);
+    std::string isPattern = ent.getStringField(CSUB_ENTITY_ISPATTERN);
 
+    EntID en;
+    if (isFalse(isPattern))
+    {
+      en.id = id;
+    }
+    else
+    {
+      en.idPattern = id;
+    }
+    en.type = type;
+
+    s->subject.entities.push_back(en);
+  }
+
+  // Condition
+  std::vector<BSONElement> conds = r.getField(CSUB_CONDITIONS).Array();
+  for (unsigned int ix = 0; ix < conds.size(); ++ix)
+  {
+    BSONObj cond = conds[ix].embeddedObject();
+    // The ONCHANGE check is needed, as a subscription could mix different conditions types in DB
+    if (std::string(cond.getStringField(CSUB_CONDITIONS_TYPE)) == "ONCHANGE")
+    {
+      std::vector<BSONElement> condValues = cond.getField(CSUB_CONDITIONS_VALUE).Array();
+      for (unsigned int jx = 0; jx < condValues.size(); ++jx)
+      {
+        std::string attr = condValues[jx].String();
+        s->subject.condition.attributes.push_back(attr);
+      }
+    }
+  }
+
+  // Note that current DB model is based in NGSIv1 and doesn't consider expressions. Thus
+  // subject.condition.expression cannot be filled. The implemetion will be enhanced once
+  // the DB model gets defined
+  // TBD
+
+}
+
+
+/* ****************************************************************************
+*
+* setNotification -
+*/
+extern void setNotification(Subscription* s, const BSONObj& r)
+{
+  // Attributes
+  std::vector<BSONElement> attrs = r.getField(CSUB_ATTRS).Array();
+  for (unsigned int ix = 0; ix < attrs.size(); ++ix)
+  {
+    std::string attr = attrs[ix].String();
+    s->notification.attributes.push_back(attr);
+  }
+
+  // Callback
+  s->notification.callback = r.getStringField(CSUB_REFERENCE);
+
+  // Throttling
+  long long throttling = r.hasField(CSUB_THROTTLING)? r.getField(CSUB_THROTTLING).numberLong() : -1;
+  if (throttling > -1)
+  {
+    // FIXME P10: int -> str conversion needed
+    s->notification.throttling.set("FIXME");
+  }
+}
+
+
+/* ****************************************************************************
+*
+* setDuration -
+*/
+extern void setDuration(Subscription* s, const BSONObj& r)
+{
+  //long long expiration = r.hasField(CSUB_EXPIRATION)? r.getField(CSUB_EXPIRATION).numberLong() : 0;
+  // FIXME P10: int -> str conversion needed
+  s->duration.set("FIXME");
```

Comment:
NTC (it will not be fixed in this PR, but in a next one; discussion is yet ongoing)